src/Microsoft.ML.LightGBM/WrappedLightGbmInterface.cs

```diff
@@ -150,6 +150,7 @@ public enum CApiDType : int
 
         [DllImport(DllName, EntryPoint = "LGBM_BoosterSaveModelToString", CallingConvention = CallingConvention.StdCall)]
         public static extern unsafe int BoosterSaveModelToString(IntPtr handle,
+            int startIteration,
```

Comment:
@guolinke -  what do you think about starting to follow https://semver.org/ with the LightGBM package? Check that site out and let me know what you think.

In general, it is surprising that there are API breaking changes in a minor update from `2.1.2.2` => `2.2.1`. Semantic versioning attempts to address these types of surprises. It really helps API customers out IMO.

Reply:
@eerhardt actually, we just released 2.2.0. 
2.2.1 is the special version for ML.Net

Reply:
Right - and in between `2.1.2.2` => `2.2.1` you made a public API breaking change.  See this change here.

Following semantic versioning, you cannot make a public API breaking change unless it is across major versions.